tests/fixtures/liteclienttest.cc

```diff
@@ -1,198 +1,67 @@
-  }
-
- private:
-  const std::string path_;
-  const std::string os_;
-  OSTreeRepoMock repo_;
-};
-
-/**
- * Class TufRepoMock
- *
- */
-class TufRepoMock {
- public:
-  TufRepoMock(const boost::filesystem::path& _root, std::string expires = "",
-              std::string correlation_id = "corellatio-id")
-      : root_{_root.string()}, repo_{_root, expires, correlation_id}, latest_{Uptane::Target::Unknown()} {
-    repo_.generateRepo(KeyType::kED25519);
-  }
-
- public:
-  const std::string& getPath() const { return root_; }
-  const Uptane::Target& getLatest() const { return latest_; }
-
-  Uptane::Target addTarget(const std::string& name, const std::string& hash, const std::string& hardware_id,
-                           const std::string& version, const Json::Value& apps_json = Json::Value()) {
-    Delegation null_delegation{};
-    Hash hash_obj{Hash::Type::kSha256, hash};
-
-    Json::Value custom_json;
-    custom_json["targetFormat"] = "OSTREE";
-    custom_json["version"] = version;
-    custom_json[Target::ComposeAppField] = apps_json;
-    repo_.addCustomImage(name, hash_obj, 0, hardware_id, "", null_delegation, custom_json);
-
-    Json::Value target_json;
-    target_json["length"] = 0;
-    target_json["hashes"]["sha256"] = hash;
-    target_json["custom"] = custom_json;
-    latest_ = Uptane::Target(name, target_json);
-    return latest_;
-  }
-
- private:
-  const std::string root_;
-  ImageRepo repo_;
-  Uptane::Target latest_;
-};
-
-/**
- * Class DeviceGatewayMock
- *
- */
-class DeviceGatewayMock {
- public:
-  static std::string RunCmd;
-
- public:
-  DeviceGatewayMock(const OSTreeRepoMock& ostree, const TufRepoMock& tuf)
-      : ostree_{ostree},
-        tuf_{tuf},
-        port_{TestUtils::getFreePort()},
-        url_{"http://localhost:" + port_},
-        req_headers_file_{tuf_.getPath() + "/headers.json"},
-        process_{RunCmd,           "--port",         port_, "--ostree", ostree_.getPath(), "--tuf-repo", tuf_.getPath(),
-                 "--headers-file", req_headers_file_} {
-    TestUtils::waitForServer(url_ + "/");
-    LOG_INFO << "Device Gateway is running on port " << port_;
-  }
-
-  ~DeviceGatewayMock() {
-    process_.terminate();
-    process_.wait_for(std::chrono::seconds(10));
-  }
-
- public:
-  std::string getOsTreeUri() const { return url_ + "/treehub"; }
-  std::string getTufRepoUri() const { return url_ + "/repo"; }
-  const std::string& getPort() const { return port_; }
-  Json::Value getReqHeaders() const { return Utils::parseJSONFile(req_headers_file_); }
-
- private:
-  const OSTreeRepoMock& ostree_;
-  const TufRepoMock& tuf_;
-  const std::string port_;
-  const std::string url_;
-  const std::string req_headers_file_;
-  boost::process::child process_;
-};
-
-std::string DeviceGatewayMock::RunCmd;
-
-
-
-/**
- * Class ClientTest
- *
- */
 class ClientTest :virtual public ::testing::Test {
  public:
   static std::string SysRootSrc;
 
  protected:
+#if defined(USE_HSM)
```

Comment:
I'd try to avoid `#if` for the HSM. For example by extending ClientTest `class ClientHSMTest: public ClientTest` then actuall test would do `class Test: public ClientHSMTest`, or `class ClientHSMTest: virtual public ::testing::Test` and then `class Test: public ClientTest, public ClientHSMTest`.
But, that's just a matter of personal taste, so that's OK for testing :).

Reply:
agree 100%:  I'd never use macros in a production environment (is really a bad smell...just like any kind of conditional statements to be honest, that is why classes exist )...but since you mention, let me take a stab at it (Even though I think it will make the code a bit more confusing)

Reply:
@mike-sul pushed implementing an interface.
I think this code makes the implicit needs for enabling HSM perhaps bit more obscure to those not familiar with the code but is more proper.